deck-service/src/main/java/magicthebuilder/deckservice/service/DeckService.java

```diff
@@ -0,0 +1,138 @@
+        if (repository.findById(id).isPresent()) {
+            Deck deck = repository.findById(id).get();
+            if (deck.getAccessLevel() == DeckAccessLevelEnum.PRIVATE) {
+                if (userId.equals(deck.getOwner().getId())) {
+                    return new DetailedDeckGetResponseDto(deck);
+                } else {
+                    throw new InaccessibleDeckException(id);
+                }
+            } else {
+                return new DetailedDeckGetResponseDto(deck);
+            }
+        } else {
+            throw new UnrecognizedDeckException(id);
+        }
+    }
+
+    public UUID addDeck(Deck deck) {
+        repository.save(deck);
+        return deck.getUuid();
+    }
+
+    public Deck createDeckFromDto(CreateDeckDto dto) {
+        Optional<User> owner = userRepository.findById(dto.getUserId());
+        if (owner.isPresent()) {
+            Deck ret = CreateDeckDto.dtoToEntityMapper().apply(dto);
+            ret.setOwner(owner.get());
+            // PLACE FOR VALIDATING DECK  //
+            return ret;
+
+        } else {
+            throw new UnrecognizedUserIdException(dto.getUserId());
+        }
+    }
+
+    public DetailedDeckGetResponseDto updateDeck(DeckUpdateRequestDto deckDto) {
+
+        Map<String, Integer> deckToAdd = new HashMap<>();
+        for (MultipleCardDto card : deckDto.getDeckCardsToAdd()) {
+            deckToAdd.put(card.getCardId(), card.getAmount());
+        }
+        Map<String, Integer> deckToRemove = new HashMap<>();
+        for (MultipleCardDto card : deckDto.getDeckCardsToRemove()) {
+            deckToRemove.put(card.getCardId(), card.getAmount());
+        }
+        Map<String, Integer> sideboardToAdd = new HashMap<>();
+        for (MultipleCardDto card : deckDto.getSideboardCardsToAdd()) {
+            sideboardToAdd.put(card.getCardId(), card.getAmount());
+        }
+        Map<String, Integer> sideboardToRemove = new HashMap<>();
+        for (MultipleCardDto card : deckDto.getSideboardCardsToRemove()) {
+            sideboardToRemove.put(card.getCardId(), card.getAmount());
+        }
+
+        Optional<Deck> optDeck = repository.findById(deckDto.getId());
+        if (optDeck.isEmpty()) {
+            throw new UnrecognizedDeckException(deckDto.getId());
+        }
+        Deck deck = optDeck.get();
+        deck.setAccessLevel(deckDto.getAccessLevel());
+        deck.setGameMode(deckDto.getGamemode());
+        deck.setName(deckDto.getName());
+        List<Card> deckCards = deck.getCards();
+        List<Card> sideboardCards = deck.getSideboard();
+        deckCards.addAll(getCardListFromMap(deckToAdd));
+        deckCards.removeAll(getCardListFromMap(deckToRemove));
+        sideboardCards.addAll(getCardListFromMap(sideboardToAdd));
+        sideboardCards.removeAll(getCardListFromMap(sideboardToRemove));
+        deck.setCards(deckCards);
+        deck.setSideboard(sideboardCards);
+        //ADD DECK VALIDATION HERE//
```

Comment:
If we want to do deck validation in deck service, Card entity would need few more fields

Reply:
Yea, this note was mostly for validating main requirements. Such as checking if deck has at least 60 cards, and if sideboard has <15 cards. We are still not sure how we will implement gamemode validation